_sc2data/Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ActorData.xml

```diff
@@ -4955,24 +5053,150 @@
     <CActorAction id="AvengerAttack2" parent="GenericAttack" effectAttack="AP_AvengerWeapon">
         <ImpactMap index="None" Model="None" Sound="DarkTemplar_AttackImpact"/>
         <ShieldFlashType value="Header"/>
     </CActorAction>
+    <CActorUnit id="AP_Baneling" parent="AP_BanelingBase" unitName="AP_Baneling">
+        <On Terms="AbilMorph.*.Finish; MorphTo AP_Baneling; MorphFrom AP_BanelingCocoon" Send="Create"/>
+        <On Terms="AbilMorph.*.Finish; MorphTo AP_Baneling; MorphFrom AP_BanelingCocoon" Send="$Birth 0 0.000000"/>
+        <On Terms="Upgrade.AP_HotSHunter.Add" Send="Create AP_HotSHunter"/>
+        <On Terms="Upgrade.AP_HotSHunter.Add" Send="Destroy"/>
+        <On Terms="ActorCreation; ValidatePlayer AP_HaveHotSHunter" Send="Create AP_HotSHunter"/>
+        <On Terms="ActorCreation; ValidatePlayer AP_HaveHotSHunter" Send="Destroy"/>
+        <On Terms="Upgrade.AP_HotsSplitterling.Add" Send="Create AP_HotSSplitterlingBig"/>
+        <On Terms="Upgrade.AP_HotsSplitterling.Add" Send="Destroy"/>
+        <On Terms="ActorCreation; ValidatePlayer AP_HaveHotSSplitterling" Send="Create AP_HotSSplitterlingBig"/>
+        <On Terms="ActorCreation; ValidatePlayer AP_HaveHotSSplitterling" Send="Destroy"/>
+    </CActorUnit>
+    <CActorMissile id="AP_HotSHunter" parent="AP_BanelingBase" unitName="AP_HotSHunter">
+        <Model value="AP_Hunter"/>
+        <BuildModel value="AP_Hunter"/>
+        <PlacementModel value="AP_Hunter"/>
+        <DeathArray index="Normal" ModelLink="AP_HunterDeath"/>
+        <DeathCustoms ModelLink="AP_HunterRuptureExplosion" Name="Variation1"/>
+        <DeathCustoms ModelLink="AP_HunterCorrosiveExplosion" Name="Variation2"/>
+        <PortraitModel value="AP_HunterPortrait"/>
+        <HeroIcon value="Assets\Textures\btn-unit-zerg-baneling-hunter.dds"/>
+        <HighlightTooltip value="Unit/Name/AP_Baneling"/>
+        <UnitIcon value="Assets\Textures\btn-unit-zerg-baneling-hunter.dds"/>
+    </CActorMissile>
+    <CActorUnit id="AP_HotSSplitterlingBig" parent="AP_BanelingBase" unitName="AP_HotSSplitterlingBig">
+        <On Terms="Upgrade.AP_HotSHunter.Add" Send="Create AP_HotSHunter"/>
+        <On Terms="Upgrade.AP_HotSHunter.Add" Send="Destroy"/>
+        <On Terms="ActorCreation; ValidatePlayer AP_HaveHotSHunter" Send="Create AP_HotSHunter"/>
+        <On Terms="ActorCreation; ValidatePlayer AP_HaveHotSHunter" Send="Destroy"/>
+        <Model value="AP_HotSSplitterlingBig"/>
+        <BuildModel value="AP_HotSSplitterlingBig"/>
+        <PlacementModel value="AP_HotSSplitterlingBig"/>
+        <DeathArray index="Normal" ModelLink="AP_SplitterlingMainDeath" SoundLink="BanelingDeathNormalSound"/>
+        <PortraitModel value="AP_SplitterlingPortrait"/>
+        <HighlightTooltip value="Unit/Name/AP_Baneling"/>
+    </CActorUnit>
+    <CActorForceSphere id="AP_BanelingAttackForce">
+        <On Terms="Effect.VolatileBurstU.Start; At Caster" Send="Create"/>

```

Comment:
AP Effect?

You  really shouldn't  create AP Actor on Vanilla Effect

Reply:
Removing the effect at all? Is it really unused in the campaign? This vanilla effect normally exists (it's a damage)

Reply:
This Effect is used only in MP, the campaign Baneling uses another set of effects to achieve the same.